include/klee/ExecutionState.h

```diff
@@ -156,7 +162,7 @@ class ExecutionState {
   ExecutionState() : ptreeNode(0) {}
 
 public:
-  ExecutionState(KFunction *kf);
+  ExecutionState(Executor *executor, KFunction *kf, bool enableNvmHeuristic=false);
```

Comment:
Some of the resolution is done at runtime, i.e. function pointers. Different states could, in theory, have different function pointers, if the configuration state is allowed to be symbolic (i.e., I have a symbolic argument that tells me whether or not to use clwb or just clflush).

The states should be pretty lightweight as most things are shared.

Reply:
I'm not sure this clarifies to me why you should be able to selectively turn on or off the heuristic for a given state.

Reply:
The heuristic itself is stateful, so it essentially part of the ExecutionState. However, since some of the analysis is non-trivially expensive, it doesn't make sense to enable it for runs where we aren't using the NVM heuristic.

This flag doesn't necessarily let you turn it on or off per state, as this constructor is only used for new execution states, i.e. running from main or another function. The other constructor clones it if it exists.

Reply:
Then I think make it a command line option and check that inside the constructor, not pass in a bool.

Reply:
Agreed, if it's something that's decided on a per-run basis, then it should be a command-line option.

Reply:
It's derived from a command line argument as is, but fine.